src/api/v1/db/oracledb/doughs-dao.js

```diff
@@ -265,4 +289,53 @@ const getDoughById = async (id) => {
   }
 };
 
-export { getDoughs, postDough, getDoughById };
+/**
+ * convert object body to an UPDATE query with correct bind params
+ *
+ * @param {object} body a PATCH request body
+ * @returns {object} `bindParams` and `query`
+ */
+const createPatchQueryAndBindParams = (body) => {
+  const inBindparams = getBindParams(body);
+
+  const query = doughsPatchQuery(inBindparams);
+
+  const bindParams = _.assign(
+    inBindparams,
+    doughsOutBindParams,
+    { id: body.data.id },
+  );
+
+  return { query, bindParams };
+};
+
+
+/**
+ * Update a specific dough recipe
+ *
+ * @param {object} body
+ * @returns {Promise<object>} the updated dough
+ */
+const updateDoughById = async (body) => {
+  if (_.isEmpty(body.data.attributes)) return getDoughById(body.data.id);
```

Comment:
since I didn't see a way in openapi 2.0 to make `attributes` have at least one of the optional values, and jsonapi [requires](https://jsonapi.org/format/#crud-updating-resource-attributes) that "any" of the attributes can be included in the request, this handles the case where someone passes in an empty `attributes` object.

Reply:
You should define what's required attributes in the openapi file. If there are no required fields in the attributes, then `{}` should be a valid PATCH body in this case.

Reply:
Yeah, in this case because we can't require any single attribute, we have to make `{}` valid. This case doesn't report an error or anything, it's just a workaround for the fact that the query we use in the rest of the function doesn't work if it had no attributes to update.